Ktisis/Interface/Components/PopupSelect.cs

```diff
@@ -0,0 +1,180 @@
+﻿using System;
+using System.Linq;
+using System.Numerics;
+using System.Collections.Generic;
+
+using ImGuiNET;
+
+namespace Ktisis.Interface.Components {
+	public static class PopupSelect {
+
+		// HoverPopupWindow Method
+		// Constants
+		private const ImGuiKey KeyBindBrowseUp = ImGuiKey.UpArrow;
+		private const ImGuiKey KeyBindBrowseDown = ImGuiKey.DownArrow;
+		private const ImGuiKey KeyBindBrowseLeft = ImGuiKey.LeftArrow;
+		private const ImGuiKey KeyBindBrowseRight = ImGuiKey.RightArrow;
+		private const ImGuiKey KeyBindBrowseUpFast = ImGuiKey.PageUp;
+		private const ImGuiKey KeyBindBrowseDownFast = ImGuiKey.PageDown;
+		private const int HoverPopupWindowFastScrollLineJump = 8; // number of lines on the screen?
+
+		// Properties
+		private static Vector2 HoverPopupWindowSelectPos = Vector2.Zero;
+		private static bool HoverPopupWindowIsBegan = false;
+		private static bool HoverPopupWindowFocus = false;
+		private static bool HoverPopupWindowSearchBarValidated = false;
+		public static int HoverPopupWindowLastSelectedItemKey = 0;
+		public static int HoverPopupWindowColumns = 1;
+		private static Action? PreviousOnClose;
+		public static int HoverPopupWindowIndexKey = 0;
+		public static dynamic? HoverPopupWindowItemForHeader = null;
+
+		[Flags]
+		public enum HoverPopupWindowFlags {
+			None = 0,
+			SelectorList = 1,
+			SearchBar = 2,
+			Grabbable = 4,
+			TwoDimenssion = 8,
+			Header = 16,
+			StayWhenLoseFocus = 32, // TODO: make it instanciable so we can have multiple
+		}
+		private static int RowFromKey(int key) => (int)Math.Floor((double)(key / HoverPopupWindowColumns));
+		private static int ColFromKey(int key) => key % HoverPopupWindowColumns;
+		private static int KeyFromRowCol(int row, int col) => (row * HoverPopupWindowColumns) + col;
+
+		public static void HoverPopupWindow(
+				HoverPopupWindowFlags flags,
+				IEnumerable<dynamic> enumerable,
+				Func<IEnumerable<dynamic>, string, IEnumerable<dynamic>> filter,
+				Action<dynamic> header,
+				Func<dynamic, bool, (bool, bool)> drawBeforeLine, // Parameters: dynamic item, bool isActive. Returns bool isSelected, bool Focus.
+				Action<dynamic> onSelect,
```

Comment:
Probably shouldn't be using dynamic here and instead use generics (if possible)

Reply:
Very good suggestion!
Done in e10daa6.